scripts/lsync.sh

```diff
@@ -3,29 +3,68 @@
 # This script checks if the English version of a page has changed since a localized
 # page has been committed.
 
-if [ "$#" -ne 1 ] || ! [ -f "$1" ]; then
-  echo -e "\nThis script checks if the English version of a page has changed since a "
-  echo -e "localized page has been committed.\n"
+if [ "$#" -lt 1 ]; then
+  echo -e "\nThis script checks if the English version of a page has changed since a " >&2
+  echo -e "localized page has been committed.\n" >&2
   echo -e "Usage:\n\t$0 <PATH>\n" >&2
   echo -e "Example:\n\t$0 content/zh/docs/concepts/_index.md\n" >&2
   exit 1
 fi
 
-LOCALIZED="$1"
-
-# Try get the English version
-EN_VERSION=`echo $LOCALIZED | sed "s/content\/..\//content\/en\//g"`
-if ! [ -f $EN_VERSION ]; then
-  echo "$EN_VERSION has been removed."
+# Validate paths
+VALID=1
+for ARG in "$@"
+do
+  if printf "%s" "$ARG" | grep -E "^/|^\.\./|^\.\.\$|/\.\./|/\.\.\$"; then
+    printf "%s: Invalid path \"%s\"\n" "$0" "$ARG" >&2
+    VALID=0
+  fi
+done
+if [ $VALID = 0 ]; then
   exit 2
 fi
 
-# Last commit for the localized file
-LASTCOMMIT=`git log -n 1 --pretty=format:%h -- $LOCALIZED`
+EXITSTATUS=0 # assume success
+for LOCALIZED in "$@"
+do
+
+  # Try get the English version
+  EN_VERSION="$( printf "%s" "$LOCALIZED" | sed "s/content\/..\//content\/en\//g" )"
+  if ! [ -f $EN_VERSION ] && [ -f "$1" ]; then
+    printf "File \"%s\" has been removed\n" "${EN_VERSION}" >&2
+    EXITSTATUS=2
+  fi
+
+  if [ -f $EN_VERSION ] && ! [ -f "$LOCALIZED" ]; then
+    printf "File \"%s\" is missing from the localization\n" "$LOCALIZED" >&2
+    if [ $EXITSTATUS -lt 2 ]; then
+      EXITSTATUS=1
+    fi
+    continue
+  fi
+
+  if ! [ -f $EN_VERSION ] && ! [ -f "$LOCALIZED" ]; then
+    printf "File \"%s\" is missing from upstream (English)\n" "$LOCALIZED" >&2
+    continue
+  fi
 
-git diff $LASTCOMMIT...HEAD $EN_VERSION
+  # Last commit for the localized file
+  LASTCOMMIT="$( git log -n 1 --pretty=format:%h -- "$LOCALIZED" )"
 
-if [ "$?" -eq 0 ]; then
-  echo "$LOCALIZED is still in sync"
-  exit 3
+  git diff --exit-code "$LASTCOMMIT...HEAD" -- "${EN_VERSION}"
```

Comment:
In general, processing multiple files (or a directory) is a different use case for a script like this. When processing a list of files, I'd expect a summary of which file has been changed. When processing a single file, I'd expect to see the detailed diff, if any.
This is not a blocking factor though.

Reply:
> When processing a single file, I'd expect to see the detailed diff, if any.
You will!

There's room to add a `--name-only` or similar option in the future.